dropwizard-jersey/src/main/java/io/dropwizard/jersey/params/NonEmptyStringParam.java

```diff
@@ -0,0 +1,21 @@
+package io.dropwizard.jersey.params;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+
+/**
+ * A parameter encapsulating optional string values with the condition that empty string inputs are
+ * interpreted as being absent. This class is useful when it is desired for empty parameters to be
+ * synonymous with absent parameters instead of empty parameters evaluating to
+ * {@code Optional.of("")}.
+ */
+public class NonEmptyStringParam extends AbstractParam<Optional<String>> {
+    protected NonEmptyStringParam(String input) {
+        super(input);
+    }
+
+    @Override
+    protected Optional<String> parse(String input) throws Exception {
+        return Optional.fromNullable(Strings.emptyToNull(input));
```

Comment:
> especially since Jersey Bean Validations could be used for this (with @NotEmpty and @NotBlank) for non-optional parameters

I think the key here is "non-optional parameters." Ideally I want to define optional parameters that if present, contain data (ie a string with a length greater than 0). If there is no data, then I want to the parameter to act as if it was absent.

I'm fine if my implementation (the last commit of this PR) isn't merged -- it's easy enough to define per project.